types.go

```diff
@@ -994,6 +1012,72 @@ func (rr *DNSKEY) String() string {
 		" " + rr.PublicKey
 }
 
+// IPSECKEY RR. See RFC 4025.
+type IPSECKEY struct {
+	Hdr         RR_Header
+	Precedence  uint8
+	GatewayType uint8
+	Algorithm   uint8
+	GatewayAddr net.IP `dns:"-"` // packing/unpacking/parsing/etc handled together with GatewayHost
+	GatewayHost string `dns:"ipsechost"`
+	PublicKey   string `dns:"base64"`
+}
+
+func (rr *IPSECKEY) String() string {
+	var gateway string
+	switch rr.GatewayType {
+	case IPSECGatewayIPv4, IPSECGatewayIPv6:
+		gateway = rr.GatewayAddr.String()
+	case IPSECGatewayHost:
+		gateway = rr.GatewayHost
+	case IPSECGatewayNone:
+		fallthrough
+	default:
+		gateway = "."
+	}
+
+	return rr.Hdr.String() + strconv.Itoa(int(rr.Precedence)) +
+		" " + strconv.Itoa(int(rr.GatewayType)) +
+		" " + strconv.Itoa(int(rr.Algorithm)) +
+		" " + gateway +
+		" " + rr.PublicKey
+}
+
+// AMTRELAY RR. See RFC 8777.
+type AMTRELAY struct {
+	Hdr               RR_Header
+	Precedence        uint8
+	DiscoveryOptional bool   `dns:"-"` // in the GatewayType byte
+	GatewayType       uint8  `dns:"amtrelaytype"`
+	GatewayAddr       net.IP `dns:"-"`
```

Comment:
why is this here and set to `dns:"-"` ?

Reply:
`AMTRELAY` and `IPSECKEY` behave very similarly with regards to the gateway, the value of `GatewayType` defines whether an IP address or hostname will be given as the gateway. Though there should be a comment here remarking on this, the same way there is one in `IPSECKEY`.

Reply:
ok, yes and no enums in Go - def. warrants a comment

Reply:
Comment added with last PR, marking as resolved.